### PR TITLE
Enable linting in stylelint-polaris and fix issues

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

Part of #5424

Lint the stylelint-polaris folder

### WHAT is this pull request doing?

 Lint the folder, and fix issues

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -3,8 +3,8 @@ node_modules
 /polaris-react/build
 /polaris-react/build-internal
 /polaris.shopify.com/.next
+/stylelint-polaris/data
```
Comment (BPScott) [inline .eslintignore:6]: Content in this folder is dynamically generated, so shouldn't be linted

Comment (aaronccasanova) [reply .eslintignore:6]: Can we add that as a comment: e.g. `# Ignored as the contents of /data is generated`

**Review by aaronccasanova — APPROVED** [review]

**Review by sam-b-rose — APPROVED** [review]
💯 ✨

File: .eslintignore (hunk 1 above)
Comment (BPScott) [reply .eslintignore:6]: It's the same for all the other build and metadata folders above, so I don't really think it's worth it - to me the default reason for inclusion in ignore files is they're generated data, and when that's not the case it's worth a comment

**Pull request merged**